src/generated/cloud/compute/v1/src/model.rs

```diff
@@ -646,6 +723,478 @@ impl wkt::message::Message for MachineTypeAggregatedList {
+        #[derive(Clone, Default, PartialEq)]
+        #[non_exhaustive]
+        pub struct Data {
+            /// [Output Only] A key that provides more detail on the warning being returned. For example, for warnings where there are no results in a list request for a particular zone, this key might be scope and the key value might be the zone name. Other examples might be a key indicating a deprecated resource and a suggested replacement, or a warning about invalid network settings (for example, if an instance attempts to perform IP forwarding but is not enabled for IP forwarding).
+            pub key: std::string::String,
+
+            /// [Output Only] A warning data value corresponding to the key.
+            pub value: std::string::String,
+
+            pub(crate) _unknown_fields: serde_json::Map<std::string::String, serde_json::Value>,
+        }
+
+        impl Data {
+            pub fn new() -> Self {
+                std::default::Default::default()
+            }
+
+            /// Sets the value of [key][crate::model::machine_type_aggregated_list::warning::Data::key].
+            pub fn set_key<T: std::convert::Into<std::string::String>>(mut self, v: T) -> Self {
+                self.key = v.into();
+                self
+            }
+
+            /// Sets the value of [value][crate::model::machine_type_aggregated_list::warning::Data::value].
+            pub fn set_value<T: std::convert::Into<std::string::String>>(mut self, v: T) -> Self {
+                self.value = v.into();
+                self
+            }
+        }
+
+        impl wkt::message::Message for Data {
+            fn typename() -> &'static str {
+                "type.googleapis.com/google.cloud.compute.v1.MachineTypeAggregatedList.warning.data"
+            }
+        }
+
+        /// The enumerated type for the [code][google.cloud.compute.v1.MachineTypeAggregatedList.warning.code] field.
+        ///
+        /// [google.cloud.compute.v1.MachineTypeAggregatedList.warning.code]: crate::model::machine_type_aggregated_list::warning::Code
+        ///
+        /// # Working with unknown values
+        ///
+        /// This enum is defined as `#[non_exhaustive]` because Google Cloud may add
+        /// additional enum variants at any time. Adding new variants is not considered
+        /// a breaking change. Applications should write their code in anticipation of:
+        ///
+        /// - New values appearing in future releases of the client library, **and**
+        /// - New values received dynamically, without application changes.
+        ///
+        /// Please consult the [Working with enums] section in the user guide for some
+        /// guidelines.
+        ///
+        /// [Working with enums]: https://google-cloud-rust.github.io/working_with_enums.html
+        #[derive(Clone, Debug, PartialEq)]
+        #[non_exhaustive]
+        pub enum Code {
```

Comment:
This enum is duplicated  184 times in the compute specification.

I wonder if we should have a mechanism to dedup this code. It is going to be 300 x 184 lines of duplicative output... 

OTOH, I expect this will be less < 5% of the total code generated for compute.

Reply:
> > mechanism to dedup
> 
> Seems risky given that there are slight variations, even if they are only in the comments (for now).

The message has differences in the comments, the `code` enum does not.  But I cannot find an internal source of truth to determine if this is an accident or because there is a single source of truth.

> > duplicative output
> 
> It is also the ergonomics. With deduping applications could have one `fn ProcessWarning(...)` vs. 184 of them.

Macros + traits + generics can take care of that, but yes, we are exporting work to our customers.